### PR TITLE
[15.0][MIG] dms_attachment_link: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT41774

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
Tested on runboat.

Can you put in the same line "Add attachments" and "Add DMS File"?

**Comment by victoralmau** [comment]
Changes done.

File: dms_attachment_link/static/src/js/dms_attachment_link.esm.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import {AttachmentBox} from "@mail/components/attachment_box/attachment_box";
+import {patch} from "web.utils";
+
+patch(
+    AttachmentBox.prototype,
+    "dms_attachment_link/static/src/js/dms_attachment_link.js",
+    {
+        _onAddDmsFile(event) {
+            event.preventDefault();
+            event.stopPropagation();
```
Comment (chienandalu) [inline dms_attachment_link/static/src/js/dms_attachment_link.esm.js]:
```suggestion
```

File: dms_attachment_link/static/src/xml/chatter.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2023 Tecnativa - Víctor Martínez
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<templates id="template" xml:space="preserve">
+    <t t-inherit="mail.AttachmentBox" t-inherit-mode="extension" owl="1">
+        <xpath expr="//button[hasclass('o_AttachmentBox_buttonAdd')]" position="after">
+            <button
+                class="o_add_dms_file_button btn btn-link"
+                type="button"
+                t-on-click="_onAddDmsFile"
```
Comment (chienandalu) [inline dms_attachment_link/static/src/xml/chatter.xml]: https://github.com/odoo/owl/blob/master/doc/reference/event_handling.md#modifiers
```suggestion
                t-on-click.stop.prevent="_onAddDmsFile"
```

File: dms_attachment_link/static/src/scss/dms_attachment_link.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.o_AttachmentBox_content {
+    text-align: center;
+    display: block;
```
Comment (chienandalu) [inline dms_attachment_link/static/src/scss/dms_attachment_link.scss]: I'd rather use bootstrap classes over the main template

image

**Review by chienandalu — CHANGES_REQUESTED** [review]
Tested :+1: Some code suggestions:

**Comment by victoralmau** [comment]
> Tested +1 Some code suggestions:

Great suggestions, thanks!

**Review by chienandalu — APPROVED** [review]
Thanks!

**Pull request merged**